src/main/java/com/google/javascript/gents/CommentLinkingPass.java

```diff
@@ -28,26 +28,32 @@
   private static final String EOL = "([ \t]*\n)?";
 
   /**
-   * These jsdocs delete everything except for the `keep` capture group Some regexes contain an
+   * These jsdocs delete everything except for the `keep` capture group. Some regexes contain an
    * empty capture group for uniform handling.
    */
   private static final Pattern[] JSDOC_REPLACEMENTS = {
+    // The <?keep> capture group is intentionally empty because it's not needed.
```

Comment:
I still don't understand why it exists at all, i.e. what would go wrong if I remove it completely? "not needed" sounds like it should be removed

It is just so that the implementation can look for the same named capturing group over any of the regexp? There is no way to handle that with with a null check or something?

Reply:
There isn't a good way to check if a capturing group exists (the two ways I've seen is catching the IllegalArgumentException or using reflection on a non-public Matcher method). I refactored a bit to get rid of <keep>.